module/userProjects/helper.js

```diff
@@ -2280,6 +2325,369 @@ module.exports = class UserProjectsHelper {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: {}
+                });
+            }
+        })
+    }
+
+    /**
+     * List user project details with certificate
+     * @method
+     * @name certificates 
+     * @param {String} userId - userId.
+     * @returns {JSON} certificate data updation details.
+    */
+
+    static certificates(userId) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                let certificateCount = 0;
+                //  get project details of user which have certificate.
+                const userProject = await projectQueries.projectDocument({
+                    userId: userId,
+                    status: CONSTANTS.common.SUBMITTED_STATUS,
+                    certificate: {$exists:true}
+                }, [
+                    "_id",
+                    "title",
+                    "status",
+                    "certificate.osid",
+                    "certificate.transactioId",
+                    "certificate.templateUrl",
+                    "certificate.status",
+                    "certificate.eligible",
+                    "certificate.message",
+                    "certificate.issuedOn"
+                ]);
+                
+                if ( !userProject.length > 0 ) {
+                    throw {
+                        status: HTTP_STATUS_CODE["bad_request"].status,
+                        message: CONSTANTS.apiResponses.PROJECT_WITH_CERTIFICATE_NOT_FOUND
+                    }
+                }
+                //  find certificate generated project count
+                for ( let projectIndex = 0; projectIndex < userProject.length; projectIndex++ ) {
+                    if ( userProject[projectIndex].certificate &&
+                         userProject[projectIndex].certificate.osid &&
+                         userProject[projectIndex].certificate.osid !== ""
+                    ) {
+                        certificateCount++;
+                    }
+                }
+                return resolve({ 
+                    success: true,
+                    message: CONSTANTS.apiResponses.PROJECTS_FETCHED,
+                    data : {
+                        data : userProject,
+                        count : userProject.length,
+                        certificateCount : certificateCount
+                    }
+
+                });
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: {}
+                });
+            }
+        })
+    }
+
+    /**
+     * Re-Issue project certificate
+     * @method
+     * @name certificateReIssue 
+     * @param {String} projectId - projectId.
+     * @returns {JSON} certificate re-issued details.
+    */
+
+     static certificateReIssue(projectId) {
```

Comment:
@VISHNUDAS-tunerlabs In project certificate object capture, when was transactionId generated, os id generated, reIssueDone date and old transaction id and certificate id

Reply:
originalTransactionInformation